spring-jdbc-todo/src/main/java/com/karan/todo/exceptions/handlers/UserExceptionHandler.java

```diff
@@ -18,6 +18,9 @@
 		HttpStatus status = null;
 
 		switch (e.getErrorCode()) {
```

Comment:
In My Opionion, this should be handled by the controller(s), and not by a exception handler. This whole handler could be deleted if you return Status + Objects in the controller.

Reply:
Well, I've kept a separate handler so that if similar exception scenario occurs in some other controller then they can simply throw the exception. They won't have to recreate the response object.

Umm... Does it sound good enough?

Reply:
I absolutely understood why you implemented a global exception handler. And I also developed APIs with global exception handler to avoid "dumb" exception messages.
But what if you software grows. At some point you'll have thousands of different cases and your global exception handlers will be a mess, because they have to switch between all these cases.

In my opinion a controller has two tasks:
1. Get the correct data (params, ....) and validate them -> if not valid, return a validation result (with all errors)
2. If the parameters are valid, call the correct implementation to continue with your job. If this failed (f. e. no db connection to save, ....) return a corresponding http error message.

So if another person looks at a controller entry point, he sees which objects and which error messages can be returned. An exception handler should not choose which http response code to return.

Don't get me wrong, it's not incorrect to use a global exception handler and you don't have to change this! I want you to give a little outlook what maybe would be better if your application grows :)

Reply:
@rennerDa I really appreciate the way you explained things. I think now I understand your point. I'll make these changes and get back to you. Thank you! :)